lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemModule.java

```diff
@@ -140,13 +140,14 @@ static TrinoFileSystemFactory createFileSystemFactory(
                 .orElseThrow(() -> new IllegalArgumentException("No factory for location: " + location));
 
         TrinoFileSystemFactory delegate = new SwitchingFileSystemFactory(loader);
+        TracingFileSystemFactory tracingFileSystemFactory = new TracingFileSystemFactory(tracer, delegate);
```

Comment:
Since `delegate` is only used here, it seems better to reassign it so that it can't accidentally be used below:
```java
delegate = new TracingFileSystemFactory(tracer, delegate);